driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java

```diff
@@ -122,26 +127,52 @@ static void validateCollation(final Connection connection, final Collation colla
     }
 
     static void validateCollation(final ConnectionDescription connectionDescription, final Collation collation) {
-        if (collation != null && !ServerVersionHelper.serverIsAtLeastVersionThreeDotFour(connectionDescription)) {
+        if (collation != null && !serverIsAtLeastVersionThreeDotFour(connectionDescription)) {
             throw new IllegalArgumentException(format("Collation not supported by wire version: %s",
                     connectionDescription.getMaxWireVersion()));
         }
     }
 
     static void validateCollationAndWriteConcern(final ConnectionDescription connectionDescription, final Collation collation,
                                                  final WriteConcern writeConcern) {
-        if (collation != null && !ServerVersionHelper.serverIsAtLeastVersionThreeDotFour(connectionDescription)) {
+        if (collation != null && !serverIsAtLeastVersionThreeDotFour(connectionDescription)) {
             throw new IllegalArgumentException(format("Collation not supported by wire version: %s",
                     connectionDescription.getMaxWireVersion()));
         } else if (collation != null && !writeConcern.isAcknowledged()) {
             throw new MongoClientException("Specifying collation with an unacknowledged WriteConcern is not supported");
         }
     }
 
+    static void validateArrayFiltersAndWriteConcern(final ConnectionDescription connectionDescription,
+                                                    final List<BsonDocument> arrayFilters, final WriteConcern writeConcern) {
+        if (arrayFilters != null) {
+            if (serverIsLessThanVersionThreeDotSix(connectionDescription)) {
+                throw new IllegalArgumentException(format("Array filters not supported by wire version: %s",
+                        connectionDescription.getMaxWireVersion()));
+            } else if (!writeConcern.isAcknowledged()) {
+                throw new MongoClientException("Specifying array filters with an unacknowledged WriteConcern is not supported");
+            }
+        }
+    }
+
+    static void validateHintAndWriteConcern(final ConnectionDescription connectionDescription, final Bson hint, final String hintString,
```

Comment:
Rename to simply validateHint.  It's confusing as currently named, as we're not validating the write concern.